### ISSUE TITLE
Development Environment Checklist for issue funders

### ISSUE THREAD

**Issue by abitrolly** [gitcoinco/web#1988]
### User Story

As a contributor I want issue projects to be easy to start, test and develop with CI support, autoreloading and coverage warnings so that I can concentrate on the specific problem that is being awarded and not on project development environment.

### Why Is this Needed

Project maintainers already know how to setup their development environment, but for other people it is unexplored and is often broken. I deal with it in my last GitHub contributions "for free", but other people may not have the time that I've got.

https://github.com/NeuroVault/NeuroVault/issues/630#issuecomment-411464043

### Current Behavior

People struggle to get projects running, tests running, metrics changes reported (coverage, ...).

### Expected Behavior

People can setup development environment in no time without (too many) hacks.

### Reproduction

Go into any gitcoin.co issue and measure the time that is needed to get project running.

### Solution

Provide a checklist for project maintainers.

* [ ] Automatic environment setup and configuration
* [ ] Easy way to run test suite in isolated environment
* [ ] Quick command to run subset of tests for this issue
* [ ] Automated code quality metrics (coverage, ...) that are seen during development
* [ ] Codebase supports live edits / reloading

What is the measured time for a newbie to setup development environment for your projects on a fresh machine?  _____ (provide virtual machine and record session)

### Definition of Done

Reports from project data collected from project checklist.

### Additional Information

More ideas.
1. Info can be provided by maintaners and contributors to compare results.
2. The checklist can be project bounty by itself to improve onboarding.

I also added provisioning of virtual machine to record session for project setup. This could be billed from ETH. Might be worth to move to separate issue once checklist is approved. I personally would like to work on this one "full time" if there is any sustainable funding.

**Comment by owocki** [comment]
this is an interesting idea....

@PixelantDesign @mbeacom @vs77bb what do yall think?

**Comment by PixelantDesign** [comment]
Great idea @abitrolly!

**Comment by frankchen07** [comment]
I feel like this would simplify the FTUX and reduce mental overhead for the returning developer.

**Comment by thelostone-mc** [comment]
- We've got pre-commit hooks to which let you run checks before committing. Additionally when you raise a PR , stickler comments on the line which needs fix and travis lists out why the build failed
- The only thing you need to do to boot up the repo is docker and a few config settings 
- defining command to run subset of tests for the issue may be a bit unrealistic -> we try and make the issue description as detailed as possible for that  


Live reload is something which maybe we could use if the community really feels it would make their life easier

**Comment by thelostone-mc** [comment]
@PixelantDesign Oh derp!  I assumed it was was our project 😂 
Thanks for pointing that out 

I get what @abitrolly & @frankchen07 are saying 
Usually, funders have a good readme `.md` file on their repo & 
from what I've seen -> 1, 2, 4, 5 usually are covered there 

3 is usually well defined in the issue

We could maybe add it aa `Tip Message` in the FTUX for a funder to remind them that more info + clarity they give on the issue == greater the probability of the issue being solved 

Making a checklist might frustrate returning funders

**Comment by abitrolly** [comment]
@kuhnchris it would be nice if you could comment on why are you closing issues. Even if it just stale issue cleanup.

**Comment by kuhnchris** [comment]
Hey there @abitrolly - if you got questions regarding closed issues - or think that they should still be active - let me know - I'm just trying to clean up and since the connected PR to this is closed, I assumed the Issue is closed. - plus as far as I can remember we do have a checklist now for funders where we explicitly tell them to provide enough infos to get started. - if it's not sufficient for you either please let me know then I'll tag it and reopen this issue - or create a new one if the requirements or situation has been updated/changed. 

Background for all of this: since we rolled out v2 or v3 of explorer and profile we changed alot of things, so there are tons of stale tickets out there, it's hard to ward through all of those, but I'm trying my best - error can happen. :-)

Thanks,
Chris

**Comment by abitrolly** [comment]
@kuhnchris I don't mind against cleaning up the tracker if if helps to see why the issue is closed, or what was the resolution. In a perfect world every resolution should come with a proof that people can validate. Being it a link to discussion decision or a screenshot of existing interface. The recipe to keep contributors happy.

**Comment by kuhnchris** [comment]
Yeah, but on the contrast: if nobody follows this up, and by that I also mean the contributor, then I cannot assume it's very important to anyone, since nobody apparently cared about it.  

Anyways, as I said, if you still find that the Funder checklist or Funder information lacks this information in the current state of the platform, open an new issue, thanks.
